Header header

LeftRight bump
Wheeldrop wheeldrop
bool wall
int16 wall_signal
Cliff cliff
bool virtual_wall
MotorOvercurrent motor_overcurrents
LeftRight dirt_detector
uint8 remote_control_command

Button buttons

int16 distance
int16 angle

Song song
Ir_Opcode opcode
uint8 dirt_detect
uint8 charger_available
uint8 open_interface_mode
uint8 oi_stream_num_packets
bool stasis

LeftRightU16 encoder_counts
LeftRight16 requested_wheel_velocity
int16 requested_velocity
int16 requested_radius

uint8 charging_state
uint16 voltage
int16 current
uint8 temperature
uint16 charge
uint16 capacity

LightBumper light_bumper

